jekyll/_includes/snippets/built-in-env-vars.adoc

```diff
@@ -1,111 +1,153 @@
 
 | `CIRCLE_PR_USERNAME`
+| GitHub, Bitbucket
 | String
 | The GitHub or Bitbucket username of the user who created the pull request. Only available on forked PRs.
 
 | `CIRCLE_PREVIOUS_BUILD_NUM`
+| GitHub, Bitbucket, GitLab
 | Integer
 | The largest job number in a given branch that is less than the current job number. **Note**: The variable is not always set, and is not deterministic. It is also not set on runner executors. This variable is likely to be deprecated, and CircleCI recommends users to avoid using it.
 
 | `CIRCLE_PROJECT_REPONAME`
+| GitHub, Bitbucket
 | String
 | The name of the repository of the current project.
 
 | `CIRCLE_PROJECT_USERNAME`
+| GitHub, Bitbucket
 | String
 | The GitHub or Bitbucket username of the current project.
 
 | `CIRCLE_PULL_REQUEST`
+| GitHub, Bitbucket
 | String
 | The URL of the associated pull request. If there are multiple associated pull requests, one URL is randomly chosen.
 
 | `CIRCLE_PULL_REQUESTS`
+| GitHub, Bitbucket
 | List
 | Comma-separated list of URLs of the current build's associated pull requests.
 
 | `CIRCLE_REPOSITORY_URL`
+| GitHub, Bitbucket
 | String
 | The URL of your GitHub or Bitbucket repository.
 
 | `CIRCLE_SHA1`
+| GitHub, Bitbucket
 | String
 | The SHA1 hash of the last commit of the current build.
 
 | `CIRCLE_TAG`
+| GitHub, Bitbucket
 | String
 | The name of the git tag, if the current build is tagged. For more information, see the <<workflows#executing-workflows-for-a-git-tag,Git Tag Job Execution section>> of the Workflows page.
 
 | `CIRCLE_USERNAME`
+| GitHub, Bitbucket, GitLab
 | String
 | The GitHub or Bitbucket username of the user who triggered the pipeline (only if the user has a CircleCI account).
 
 | `CIRCLE_WORKFLOW_ID`
+| GitHub, Bitbucket, GitLab
 | String
 | A unique identifier for the workflow instance of the current job. This identifier is the same for every job in a given workflow instance.
 
 | `CIRCLE_WORKFLOW_JOB_ID`
+| GitHub, Bitbucket, GitLab
 | String
 | A unique identifier for the current job.
 
 | `CIRCLE_WORKFLOW_WORKSPACE_ID`
+| GitHub, Bitbucket, GitLab
 | String
 | An identifier for the <<glossary#workspace,workspace>> of the current job. This identifier is the same for every job in a given workflow.
 
 | `CIRCLE_WORKING_DIRECTORY`
+| GitHub, Bitbucket, GitLab
 | String
 | The value of the `working_directory` key of the current job.
 
 | `CIRCLE_INTERNAL_TASK_DATA`
+| GitHub, Bitbucket, GitLab
 | String
 | **Internal**. A directory where internal data related to the job is stored. We do not document the contents of this directory; the data schema is subject to change.
 |===
+
+If you must use the environment variables that are deprecated for GitLab SaaS in your GitLab pipelines, you can do so by using the xref:env-vars#environment-variable-usage-options[`environment` key] in your configuration and providing your own mappings with suitable pipeline values:
```

Comment:
Maybe something like the following to be specific about what the example is hwoing? Or could add comments to the snippet?

For GitLab projects, if you need the functionality provided by these deprecated env vars, you can create the same functionality using pipeline values in your configuration file. The example below shows polulating the CIRCLE_PROJECT_REPONAME env var using the pipeline.trigger_parameters.gitlab.repo_name pipeline value